### PR TITLE
fix: Updated java/custom-resource to use non-deprecated software.amazon.awscdk.CustomResource construct

### LINKED ISSUE #756: feat(java): Replace the custom resource example with code that does not use a deprecated construct
### Describe the feature

The [java/custom-resource](https://github.com/aws-samples/aws-cdk-examples/tree/master/java/custom-resource) example should be replaced since the construct that it uses, [CustomResourceProvider](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-cloudformation.CustomResourceProvider.html) is deprecated as of v2.

### Use Case

"As a CDK developer I want to learn cdk from the aws-cdk-examples repo which demonstrates the most relevant, updated code possible, so that I can define and deploy IaC that is maintainable and reliable"

### Proposed Solution

I propose rewriting the code in the [java/custom-resource](https://github.com/aws-samples/aws-cdk-examples/tree/master/java/custom-resource) to use the [software.amazon.awscdk.CustomResource](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.CustomResource.html) construct since it is supported in CDK v2.

### Other Information

The [CustomResourceProvider](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-cloudformation.CustomResourceProvider.html) construct that is used in the example is so outdated, that it only supports Node.js Lambda code.

Fixing this should also address bugs #746, #748

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### Language

Java

### PR DESCRIPTION
**Pull request by its-mirus-lu**
Updated Java custom-resource example by refactoring it to use the software.amazon.awscdk.CustomResource construct. This also brings this example in line with the Typsecript example in this repo.

This update will also address bugs #746, #748

Fixes #756, #746, #748

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

File: java/custom-resource/lambda/custom-resource-handler.py [hunk 1]
```diff
@@ -1,26 +1,42 @@
...
+  print(event)
+  request_type = event['RequestType']
+  if request_type == 'Create': return on_create(event)
+  if request_type == 'Update': return on_update(event)
+  if request_type == 'Delete': return on_delete(event)
+  raise Exception("Invalid request type: %s" % request_type)
+
+def on_create(event):
+  props = event["ResourceProperties"]
+  print("create new resource with props %s" % props)
```
Comment (Naumel) [inline java/custom-resource/lambda/custom-resource-handler.py:11]:
```suggestion
  print("Create new resource with props %s" % props)
```

File: java/custom-resource/lambda/custom-resource-handler.py [hunk 2]
```diff
@@ -1,26 +1,42 @@
...
+
+  message = event['ResourceProperties']['Message']
+
+  attributes = {
+      'Response': 'You said "%s"' % message
```
Comment (Naumel) [inline java/custom-resource/lambda/custom-resource-handler.py:16]:
```suggestion
      'Response': 'Resource message "%s"' % message
```

File: java/custom-resource/lambda/custom-resource-handler.py [hunk 3]
```diff
@@ -1,26 +1,42 @@
...
+  }
+  return { 'Data': attributes }
+
+def on_update(event):
+  physical_id = event["PhysicalResourceId"]
+  props = event["ResourceProperties"]
+  print("update resource %s with props %s" % (physical_id, props))
```
Comment (Naumel) [inline java/custom-resource/lambda/custom-resource-handler.py:23]:
```suggestion
  print("Update resource %s with props %s" % (physical_id, props))
```

File: java/custom-resource/lambda/custom-resource-handler.py [hunk 4]
```diff
@@ -1,26 +1,42 @@
...
+  # ...
+
+  return { 'PhysicalResourceId': physical_id }
+
+def on_delete(event):
+  physical_id = event["PhysicalResourceId"]
+  print("delete resource %s" % physical_id)
```
Comment (Naumel) [inline java/custom-resource/lambda/custom-resource-handler.py:30]:
```suggestion
  print("Delete resource %s" % physical_id)
```

File: java/custom-resource/lambda/custom-resource-handler.py [hunk 5]
```diff
@@ -1,26 +1,42 @@
...
+  # ...
+
+  return { 'PhysicalResourceId': physical_id }
+
+def is_complete(event, context):
+  physical_id = event["PhysicalResourceId"]
+  request_type = event["RequestType"]
+
+  # check if resource is stable based on request_type
+  # is_ready = ...
```
Comment (Naumel) [inline java/custom-resource/lambda/custom-resource-handler.py:40]: Left over?

**Review by Naumel — CHANGES_REQUESTED** [review]
Hello, thank you for your contribution. A few of the comments are nit picks, but it looks like there is a case to be made for some code clean-up too.

File: java/custom-resource/lambda/custom-resource-handler.py (hunk 1 above)
Comment (its-mirus-lu) [reply java/custom-resource/lambda/custom-resource-handler.py:11]: Addressed in new commit.

File: java/custom-resource/lambda/custom-resource-handler.py (hunk 2 above)
Comment (its-mirus-lu) [reply java/custom-resource/lambda/custom-resource-handler.py:16]: Addressed in new commit.

File: java/custom-resource/lambda/custom-resource-handler.py (hunk 4 above)
Comment (its-mirus-lu) [reply java/custom-resource/lambda/custom-resource-handler.py:30]: Addressed in new commit.

**Comment by its-mirus-lu** [comment]
@Naumel: I've updated the PR with your recommendations. Please have a look when you've got a moment.

**Review by kaizio — APPROVED** [review]

**Pull request merged**